assembly/src/parsers/u32_ops.rs

```diff
@@ -161,16 +161,31 @@ pub fn parse_u32add(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), Ass
 
 /// Translates u32addc assembly instruction to VM operations.
 ///
-/// In the unsafe mode this translates directly to `U32ADDC` operation. In the safe mode,
-/// we also assert that both inputs are u32 values.
+/// In the unsafe mode this translates directly to `U32ADD3` operation. In the safe mode,
+/// we also assert that all three inputs are u32 values.
 ///
 /// VM cycles per mode:
-/// - u32addc: 8 cycles
+/// - u32addc: 4 cycles
 /// - u32addc.unsafe: 1 cycle
-pub fn parse_u32addc(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
-    handle_u32_and_unsafe_check(span_ops, op, false)?;
+pub fn parse_u32add3(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+    // handle_u32_and_unsafe_check(span_ops, op, false)?;
+
+    match op.num_parts() {
+        0 => return Err(AssemblyError::missing_param(op)),
+        1 => {
+            span_ops.push(Operation::U32assert2);
+            span_ops.push(Operation::MovUp2);
+            span_ops.push(Operation::U32assert2);
+        }
+        2 => {
+            if op.parts()[1] != "unsafe" {
+                return Err(AssemblyError::invalid_param(op, 1));
+            }
+        }
```

Comment:
nit: you might be able to get rid of curly brackets around the "if" block.

Reply:
`rustfmt` check is failing if I try to remove it. When I run the following command, it automatically introduces those braces.
`cargo +nightly fix --allow-staged --allow-dirty --all-targets --all-features && cargo +nightly fmt && cargo +nightly clippy`

Reply:
Ah! Let's change it back to make rustfmt happy then.